pandas_datareader/data.py

```diff
@@ -25,35 +25,43 @@
 
 
 def get_data_fred(*args, **kwargs):
-    return FredReader(*args, **kwargs).read()
+    with FredReader(*args, **kwargs) as reader:
+        return reader.read()
 
 
 def get_data_famafrench(*args, **kwargs):
-    return FamaFrenchReader(*args, **kwargs).read()
+    with FamaFrenchReader(*args, **kwargs) as reader:
+        return reader.read()
 
 
 def get_data_google(*args, **kwargs):
-    return GoogleDailyReader(*args, **kwargs).read()
+    with GoogleDailyReader(*args, **kwargs) as reader:
+        return reader.read()
 
 
 def get_data_yahoo(*args, **kwargs):
-    return YahooDailyReader(*args, **kwargs).read()
+    with YahooDailyReader(*args, **kwargs) as reader:
```

Comment:
Don't need to test all readers.  Lets  stick to Fred and FamaFrench since these are likely to work.